api/src/test/java/com/nicknathanjustin/streamercontracts/votes/VoteServiceImplTest.java

```diff
@@ -49,71 +50,97 @@ public void recordVote_nullContractModel_throwsException() {
+    @Test
+    public void recordVote_contractNotAccepted_returnsFalse() {
+        boolean voteSucceeded = voteServiceImpl.recordVote(createUserModel(PROPOSER_ID),
+                createContractModel(getValidContractModelBuilder().state(ContractState.OPEN.name()),
                         createUserModel(PROPOSER_ID),
                         createUserModel(STREAMER_ID)),
                 true);
+        Assert.assertFalse(voteSucceeded);
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void recordVote_contractNotAccepted_throwsException() {
-        voteServiceImpl.recordVote(createUserModel(PROPOSER_ID),
-                createContractModel(getValidContractModelBuilder().isAccepted(false),
+    @Test
+    public void recordVote_contractIsCompleted_returnsFalse() {
+        boolean voteSucceeded = voteServiceImpl.recordVote(createUserModel(PROPOSER_ID),
+                createContractModel(
+                        getValidContractModelBuilder().state(ContractState.COMPLETED.name()),
                         createUserModel(PROPOSER_ID),
                         createUserModel(STREAMER_ID)),
                 true);
+        Assert.assertFalse(voteSucceeded);
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void recordVote_contractIsCompleted_throwsException() {
-        voteServiceImpl.recordVote(createUserModel(PROPOSER_ID),
+    @Test
+    public void recordVote_contractIsPastSettlesTimestamp_returnsFalse() {
+        boolean voteSucceeded = voteServiceImpl.recordVote(createUserModel(PROPOSER_ID),
                 createContractModel(
-                        getValidContractModelBuilder().isCompleted(true),
+                        getValidContractModelBuilder().settlesAt(SETTLE_CONTRACT_TIME_STAMP),
                         createUserModel(PROPOSER_ID),
                         createUserModel(STREAMER_ID)),
-                true);
+                        true);
+        Assert.assertFalse(voteSucceeded);
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void isVotingComplete_contractIsCommunityContract_throwsException() {
-        voteServiceImpl.isVotingComplete(createContractModel(
+    @Test
+    public void isVotingComplete_contractIsCommunityContract_returnsFalse() {
+        final ContractModel contractModel = createContractModel(
                 getValidContractModelBuilder().isCommunityContract(true),
                 createUserModel(PROPOSER_ID),
-                createUserModel(STREAMER_ID))
+                createUserModel(STREAMER_ID)
         );
+
+        final UUID contractId = contractModel.getId();
+        final Optional<VoteModel> optionalProposerVote = voteServiceImpl.getVoteByContractIdAndVoterId(contractId, contractModel.getProposer().getId());
+        final Optional<VoteModel> optionalStreamerVote = voteServiceImpl.getVoteByContractIdAndVoterId(contractId, contractModel.getStreamer().getId());
```

Comment:
Lets talk offline about this. Its an anti-pattern for unit tests to test multiple methods.

Reply:
Fixed